controllers/clustertopology_controller_sync.go

```diff
@@ -16,10 +16,69 @@ limitations under the License.
 
 package controllers
 
-import "context"
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
 // reconcileState reconciles the current and desired state of the managed Cluster topology.
 func (r *ClusterTopologyReconciler) reconcileState(ctx context.Context, current, desired *clusterTopologyState) error {
-	// TODO: add reconcile logic
 	return nil
 }
+
+type twoWaysMergePatchHelper struct {
+	client   client.Client
+	original client.Object
+	data     []byte
+}
+
+// createTwoWaysMergePatchHelper will return a patch that yields the modified document when applied to the original document.
+// NOTE: In the case of ClusterTopologyReconciler, original is the current object, modified is the desired object, and
+// the patch returns all the changes required to align current to what is defined in desired; fields not defined in desired
+// are going to be preserved without changes.
+func createTwoWaysMergePatchHelper(original, modified client.Object, c client.Client) (*twoWaysMergePatchHelper, error) {
```

Comment:
```suggestion
func createTwoWayMergePatchHelper(original, modified client.Object, c client.Client) (*twoWaysMergePatchHelper, error) {
```
nit (I think, I'm not 100% sure)
Would also apply at least to go doc, struct and error message below